gobblin-modules/gobblin-kafka-common/src/main/java/gobblin/kafka/client/AbstractBaseKafkaConsumerClient.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package gobblin.kafka.client;
+
+import java.util.List;
+
+import com.google.common.base.Preconditions;
+import com.typesafe.config.Config;
+
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.util.ConfigUtils;
+
+
+/**
+ * A base {@link GobblinKafkaConsumerClient} that sets configurations shared by all {@link GobblinKafkaConsumerClient}s
+ */
+public abstract class AbstractBaseKafkaConsumerClient implements GobblinKafkaConsumerClient {
```

Comment:
getFilteredTopics() should be movable to this class. Same applies for some of the metadata handling. The actual read of the topics metadata should be implementation-specific but there is additional post-processing that should be shareable.